mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,786 @@
+                          _mlflowdbfs_path, _should_use_mlflowdbfs)
+from mlflow.store.artifact.databricks_artifact_repo import \
+    DatabricksArtifactRepository
+from mlflow.tracking._model_registry import DEFAULT_AWAIT_MAX_SLEEP_SECONDS
+from mlflow.tracking.artifact_utils import (_download_artifact_from_uri,
+                                            _get_root_uri_and_artifact_path)
+from mlflow.utils import databricks_utils
+from mlflow.utils.docstring_utils import LOG_MODEL_PARAM_DOCS, format_docstring
+from mlflow.utils.environment import (_CONDA_ENV_FILE_NAME,
+                                      _CONSTRAINTS_FILE_NAME,
+                                      _PYTHON_ENV_FILE_NAME,
+                                      _REQUIREMENTS_FILE_NAME,
+                                      _mlflow_conda_env, _process_conda_env,
+                                      _process_pip_requirements, _PythonEnv)
+from mlflow.utils.file_utils import (TempDir,
+                                     shutil_copytree_without_file_permissions,
+                                     write_to)
+from mlflow.utils.model_utils import (_add_code_from_conf_to_system_path,
+                                      _get_flavor_configuration_from_uri,
+                                      _validate_and_copy_code_paths)
+from mlflow.utils.uri import (append_to_uri_path, dbfs_hdfs_uri_to_fuse_path,
+                              generate_tmp_dfs_path,
+                              get_databricks_profile_uri_from_artifact_uri,
+                              is_local_uri, is_valid_dbfs_uri)
+
+FLAVOR_NAME = "johnsnowlabs"
+_JOHNSNOWLABS_ENV_VARS = ('SECRET', 'AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'SPARK_NLP_LICENSE')
+_JOHNSNOWLABS_MODEL_PATH_SUB = "jsl-model"
+_MLFLOWDBFS_SCHEME = "mlflowdbfs"
+_SPARK_NLP_JAR_URI = f"https://s3.amazonaws.com/auxdata.johnsnowlabs.com/public/jars/spark-nlp-assembly-{settings.raw_version_nlp}.jar"
+_SPARK_NLP_JSL_JAR_URI = "https://pypi.johnsnowlabs.com/{secret}/spark-nlp-jsl-" + f"{settings.raw_version_medical}.jar"
+_SPARK_NLP_JSL_WHEEL_URI = "https://pypi.johnsnowlabs.com/{secret}/spark-nlp-jsl/spark_nlp_jsl-" + f"{settings.raw_version_medical}-py3-none-any.whl"
+_logger = logging.getLogger(__name__)
+
+
+def _validate_env_vars():
+    if not all(os.environ.get(var) for var in _JOHNSNOWLABS_ENV_VARS):
+        raise Exception(
+            f'Please set {",".join(var for var in _JOHNSNOWLABS_ENV_VARS if not os.environ.get(var))}'
+            f' environment variables to your secret key before saving or logging your model')
+
+
+def _download_url(url, save_path):
+    file_name = url.split("/")[-1]
+    save_path = Path(save_path) / file_name
+    # Download the file from `url` and save it locally under `file_name`:
+    with urlopen(url) as response, open(str(save_path), "wb") as out_file:
+        shutil.copyfileobj(response, out_file)
+
+
+def _download_sparknlp_jar(out_folder):
```

Comment:
Where is this function used?

Reply:
not at all, was actually a placeholder now it's dropped thanks
resolved in https://github.com/mlflow/mlflow/pull/8556/commits/76d55dcdd647b6d183e933e3f7802995c95fc2fc